doc/source/whatsnew/v0.19.0.txt

```diff
@@ -972,3 +972,5 @@ Bug Fixes
 
 - Bug in ``Index`` raises ``KeyError`` displaying incorrect column when column is not in the df and columns contains duplicate values (:issue:`13822`)
 - Bug in ``Period`` and ``PeriodIndex`` creating wrong dates when frequency has combined offset aliases (:issue:`13874`)
+
+- Bug where files opened by parsers (used by `pd.read_csv`, e.g.) were not closed properly. (:issue:`13940`)
```

Comment:
It does not re-open after each iteration AFAIK.  You can see for yourself by writing to a CSV file, opening it with `read_csv` and passing in `chunksize`.  Try then deleting the file in a separate window.  You should be able to do so.